scripts/build/Platform/Linux/deploy_cdk_applications.sh

```diff
@@ -80,11 +80,14 @@ fi
 
 # Set temporary AWS credentials from the assume role
 credentials=$(aws sts assume-role --query Credentials.[SecretAccessKey,SessionToken,AccessKeyId] --output text --role-arn $ASSUME_ROLE_ARN --role-session-name o3de-Automation-session)
-AWS_SECRET_ACCESS_KEY=$(echo "$credentials" | cut -d' ' -f1)
-AWS_SESSION_TOKEN=$(echo "$credentials" | cut -d' ' -f2)
-AWS_ACCESS_KEY_ID=$(echo "$credentials" | cut -d' ' -f3)
+export AWS_SECRET_ACCESS_KEY=$(echo $credentials | cut -d' ' -f1)
+export AWS_SESSION_TOKEN=$(echo $credentials | cut -d' ' -f2)
+export AWS_ACCESS_KEY_ID=$(echo $credentials | cut -d' ' -f3)
 
-O3DE_AWS_DEPLOY_ACCOUNT=$(echo "$ASSUME_ROLE_ARN" | cut -d':' -f5)
+export O3DE_AWS_DEPLOY_ACCOUNT=$(echo "$ASSUME_ROLE_ARN" | cut -d':' -f5)
+if [[ -z "$O3DE_AWS_PROJECT_NAME" ]]; then
+   export O3DE_AWS_PROJECT_NAME=$BRANCH_NAME-$PIPELINE_NAME-Linux
```

Comment:
Minor: change requires here? if branch name and pipeline name are empty

Reply:
Deployment will fail if branch name and pipeline name are empty and cdk will throw an error about the invalid project. These are Jenkins environment variables shared by all the Jenkins job.

Reply:
I remember for local testing, you expect to run this script as well?

Reply:
Yeah, for local testing, our instruction is to set the O3DE_AWS_PROJECT_NAME environment variable and doesn't mention about the Jenkins parameters (BRANCH_NAME and PIPELINE_NAME) at all. As Allie commented before, these two environment variables are confusing for local testing. My thought here is that we use O3DE_AWS_PROJECT_NAMEif it exists (We can set it on Jenkins as well if required in the future) and use $BRANCH_NAME-$PIPELINE_NAME-Platform as the backup AWS project name which is only for Jenkins.

Reply:
sounds good to me, just want to confirm behavior, as previously you don't need to set this env variable for local testing as well, it will use AWSAUTO if variable is empty?

Reply:
Previously user need to set O3DE_AWS_PROJECT_NAME=AWSAUTO for local testing. our python test used AWSAUTO as the  default value if this environment variable is not set.

Reply:
sorry, it confuses me now.
based on previous code, python test tries to get O3DE_AWS_PROJECT_NAME, if this env variable doesn't exist, it will use AWSAUTO as default value. Then why we have to set O3DE_AWS_PROJECT_NAME for local testing?

Reply:
Synced offline: the previous O3DE_AWS_PROJECT_NAME is used to make sure that AWSI automation tests use the stacks deployed by the CDK applications.